packages/wdio-config/src/lib/ConfigParser.js

```diff
@@ -68,6 +68,7 @@ export default class ConfigParser {
      */
     merge (object = {}) {
         this._config = merge(this._config, object, MERGE_OPTIONS)
+        let spec = object.spec ? object.spec : []
```

Comment:
let's check if `spec` is an array too:
```js
const spec = Array.isArray(object.spec) ? object.spec : []
```

Reply:
Added the isArray check there and to two other places.